editoast/editoast_schemas/src/infra/direction.rs

```diff
@@ -0,0 +1,24 @@
+use editoast_common::rangemap_utils::TravelDir;
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+editoast_common::schemas! {
+    Direction,
+}
+
+#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq, Eq, Hash, ToSchema)]
+#[serde(deny_unknown_fields, rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Direction {
+    StartToStop,
+    StopToStart,
+}
+
+impl From<Direction> for TravelDir {
+    fn from(val: Direction) -> TravelDir {
+        match val {
+            Direction::StartToStop => TravelDir::StartToStop,
+            Direction::StopToStart => TravelDir::StopToStart,
+        }
+    }
+}
```

Comment:
I would say this should stay defined next to `TravelDir`, not in the schema since `TravelDir` has no existence in the RailJSON, right? I'd be happy to have another opinion on this (cc @leovalais).

Reply:
I need `Direction` in `editoast_common` for this `From` (Clippy doesn't accept `Into`) in this case.
`editoast_schemas' already depends on `editoast_common`.

Reply:
I'd say the problem here is that because of its name, `TravelDir` seems to be a railjson object when it's not. It's used to shift a range map (cf. rangemap_utils.rs). However `Direction` is indeed a railjson object (whilst generic enough to be placed in `editoast_schemas::prelude` I'd say).

I'd suggest:

1. rename `TravelDir` to `Direction` with variants `StartToEnd` and `EndToStart` (to match the naming of the crate `rangemap`)
2. rename `travel_range_map` to `fn shift_range_map(range_map, shifting_point, direction)`
3. `impl From<Direction> for editoast_common::rangemap_utils::TravelDir`, that way we're making it clear it's two very different things

Reply:
I'll fully review this PR tomorrow